### PR TITLE
Remove Fedora 23 support

### PR DESCRIPTION
**Pull request by gkhanna79**
CoreFX is no longer producing Fedora 23 packages, so remove the support for the same to address build breaks.

@eerhardt @weshaggard PTAL

### CODE REVIEW

**Comment by gkhanna79** [comment]
@mmitche PTAL as Groovy change

**Comment by mmitche** [comment]
LGTM for groovy.

File: netci.groovy [hunk 1]
```diff
@@ -10,7 +10,7 @@ def project = GithubProject
 def branch = GithubBranchName
 def isPR = true
 
-def platformList = ['Debian8.2:x64:Debug', 'PortableLinux:x64:Release', 'Ubuntu:arm:Release', 'Ubuntu:x64:Release', 'Ubuntu16.04:arm:Release', 'Ubuntu16.04:x64:Release', 'Ubuntu16.10:x64:Release', 'OSX10.12:x64:Release', 'Windows_NT:x64:Release', 'Windows_NT:x86:Debug', 'Windows_NT:arm:Debug', 'RHEL7.2:x64:Release', 'CentOS7.1:x64:Debug', 'Fedora23:x64:Debug', 'OpenSUSE42.1:x64:Debug']
+def platformList = ['Debian8.2:x64:Debug', 'PortableLinux:x64:Release', 'Ubuntu:arm:Release', 'Ubuntu:x64:Release', 'Ubuntu16.04:arm:Release', 'Ubuntu16.04:x64:Release', 'Ubuntu16.10:x64:Release', 'OSX10.12:x64:Release', 'Windows_NT:x64:Release', 'Windows_NT:x86:Debug', 'Windows_NT:arm:Debug', 'RHEL7.2:x64:Release', 'CentOS7.1:x64:Debug', 'OpenSUSE42.1:x64:Debug']
```
Comment (eerhardt) [inline netci.groovy]: How come fedora 24 isn't in this list?

Comment (gkhanna79) [inline netci.groovy]: I am trying to clean minimal set that unblocks the build. Eventually, we want to clean a bunch of the builds, including 42.1...

Comment (eerhardt) [inline netci.groovy]: Can you log an issue?

Comment (gkhanna79) [inline netci.groovy]: https://github.com/dotnet/core-setup/issues/1913

**Comment by danmosemsft** [comment]
Thanks for fixing this @gkhanna79  -- sorry, I was out this morning. I don't know how this works together so I just relied on code review in the change that broke this.

I did add a cleanup job https://github.com/dotnet/corefx/pull/17753

**Pull request merged**